src/app/data-client.ts

```diff
@@ -52,12 +52,26 @@ export class DataClient {
       if (!dataList || dataList.length === 0) {
         break;
       }
+      const mdListLength = response.getMetadataList().length;
+
       dataArray.push(
-        ...dataList.map((data) => ({
-          ...data.toObject(),
-          timeRequested: data.getTimeRequested()?.toDate(),
-          timeReceived: data.getTimeReceived()?.toDate(),
-        }))
+        ...dataList.map((data) => {
+          const mdIndex = data.getMetadataIndex();
+          /*
+           * The returned metadata might be empty if the metadata index of
+           * the data is out of the bounds of the returned metadata list
+           */
```

Comment:
this should be at the top of the method like https://github.com/viamrobotics/viam-typescript-sdk/blob/a886344f35c1593647c120a110fb266ceb356eb8/src/types.ts#L12-L16, so it'll show up in ts.viam.dev